jvm-packages/xgboost4j-spark/src/main/scala/ml/dmlc/xgboost4j/scala/spark/XGBoost.scala

```diff
@@ -257,21 +266,19 @@ object XGBoost extends Serializable {
    * @param useExternalMemory indicate whether to use external memory cache, by setting this flag as
    *                          true, the user may save the RAM cost for running XGBoost within Spark
    * @param missing the value represented the missing value in the dataset
-   * @param baseMargin initial prediction for boosting.
    * @throws ml.dmlc.xgboost4j.java.XGBoostError when the model training is failed
    * @return XGBoostModel when successful training
    */
   @throws(classOf[XGBoostError])
   def trainWithRDD(
-      trainingData: RDD[MLLabeledPoint],
+      trainingData: RDD[XGBLabeledPoint],
```

Comment:
I thought about this and came up with three possible solutions:

1. Allow training with both `RDD[MLLabeledPoint]` and `XGBLabeledPoint` and duplicate `trainWithRDD`. The users who need `baseMargin` would use `XGB*` data type, while the others would just go with the one from Spark ML. The benefit is obvious: no API changes on the Spark side. The drawback though is code duplication.
2. Support only `RDD[XGBLabeledPoint]` but provide an implicit conversion `RDD[XGBLabeledPoint]->RDD[MLLabeledPoint]`. This would also introduce no noticeable changes to the API, but the trick is to make the conversion available to the compiler without any additional imports. IIUC the only way to make it work is to put the conversion inside the definition of `XGBLabeledPoint` which is not possible, since this class is defined in `xgboost4j.
3. Support only `RDD[XGBLabeledPoint]` but give the users a way to convert from `MLLabeledPoint` explicitly. This is what's implemented in this PR and the drawback is surely a breaking API change. Moreover, the users would have to import implicits/migrate their code to make it work. 

Which way do you think we should go?

N.B. To my knowledge, there is no way to support `baseMargin` other than pass it along with other data in the same RDD. I've tried to make it work with a separate `RDD[Float]` but failed since it seems to be no way to force exactly the same partitioning (in terms of partition sizes) for two unrelated RDDs.